### ISSUE TITLE
GameRules#setOrCreateGameRule -> set

### ISSUE THREAD

**Issue by InsomniaKitten** [ModCoderPack/MCPBot-Issues#699]
Previously, there used to be a setter, and a setter that created a new gamerule for the setter key if one was not present. Now we only have a setter - and the MCP name has been transferred over incorrectly.

- [ ] `func_82764_b` in `net.minecraft.world.GameRules`: `setOrCreateGameRule` -> `set`

```java
public void setOrCreateGameRule(String key, String ruleValue, @Nullable MinecraftServer server) {
	GameRules.Value gamerules$value = this.rules.get(key);

	if (gamerules$value != null) {
		gamerules$value.setValue(ruleValue, server);
	}
}
```

**Comment by PaleoCrafter** [comment]
`set` strongly implies that the effect will always be performed. The method will only succeed if the rule exists at all, however, so the name should reflect that. Something like `setIfPresent` would be better, imo.

**Comment by kashike** [comment]
Choosing an ugly name for a method instead of adding javadocs is silly:
- `set`: `Sets the value of {@code key} if a rule has been defined for that key.`

**Comment by Runemoro** [comment]
> Choosing an ugly name for a method instead of adding javadocs is silly

No, it isn't. It's better if the name clearly describes what it does without having to look at javadocs. If the method was named `set`, many people would probably not even look at the javadocs for such an obviously named method, and wrongly think that the method will add the rule if it's not present.

And `setIfPresent` isn't an "ugly name", the same way `Map.putIfPresent` isn't. Mojang could decide to add a `set` method later on that adds the rule if it's not present, which would require using `setIfPresent`.

**Comment by InsomniaKitten** [comment]
The difference between this and something like the methods in `Map`, is that there is only one setter. There are not multiple setters each with differing semantics. `Map`'s` putIfAbsent` is required because it already has `put`, not because it is beneficial to have the verbosity. It is named as so to seperate it from the existing `put` method. I agree that a javadoc would be fine to describe the semantics of the setter, in much the same way `Set#add` describes the semantics of the `boolean` return value in its javadoc.
